core/hw/naomi/m4cartridge.h

```diff
@@ -42,7 +42,7 @@ class M4Cartridge: public NaomiCartridge {
 	bool Read(u32 offset, u32 size, void *dst) override;
 	bool Write(u32 offset, u32 size, u32 data) override;
 
-	void* GetDmaPtr(u32 &size) override;
+	void* GetDmaPtr(u32 &limit) override;
```

Comment:
I would prefer to use `size` for consistency with most other cart types. It's also more meaningful for users of the interface.